v2/tools/asoctl/internal/importing/resource_importer.go

```diff
@@ -159,6 +159,51 @@ func (ri *ResourceImporter) Import(ctx context.Context) (*ResourceImportResult,
 	}, nil
 }
 
+func (ri *ResourceImporter) ImportResource(ctx context.Context, rsrc ImportableResource) error {
+	// Import it
+	gk := rsrc.GroupKind()
+	name := fmt.Sprintf("%s %s", gk, rsrc.Name())
+	bar := ri.progress.AddBar(
+		0, // zero total because we don't know how much work will be needed
+		mpb.BarRemoveOnComplete(),
+		mpb.PrependDecorators(
+			decor.Name(name, decor.WCSyncSpaceR)),
+		mpb.AppendDecorators(decor.Percentage(decor.WC{W: 5})))
+
+	defer func() {
+		// Ensure the progress bar is always updated when we're done
+		// We're done with this resource, so remove it from the progress bar
+		bar.Increment()
+		bar.SetTotal(bar.Current(), true)
+	}()
+
+	pending, err := rsrc.Import(ctx, bar)
+	if err != nil {
+		var skipped *ImportSkippedError
+		if errors.As(err, &skipped) {
+			ri.log.V(1).Info(
+				"Skipped",
+				"kind", gk,
+				"name", rsrc.Name(),
+				"because", skipped.Because)
+			return nil
+		}
+
+		ri.log.Error(err,
```

Comment:
Do we need to log the error here and also return an error?

Will this error be logged above (resulting in a duplicate error log?)

Reply:
Want to log it here so the presence of errors can be seen during the run, and in the right context.

Also need to return an error to ensure `asoctl` doesn't exit with a success code. 

Maybe need some fine tuning on what's in each message?